Automattic/wp-calypso
# Donation block: Make content editable

#### Changes proposed in this Pull Request

Adds the ability to edit the default content included in a donations block.

![Jul-02-2020 16-28-00](https://user-images.githubusercontent.com/1233880/86372250-3a69bd00-bc82-11ea-951d-ed57b0c1ebd9.gif)

#### Testing instructions

* Generate a new FSE build and sync it to your sandbox: `cd apps/full-site-editing; yarn dev --sync`.
* Sandbox the API and the store. More info in PCYsg-lW4-p2 #sandbox-method.
* Sandbox a test site with a paid plan, and ensure the site has the `fse-donations-block` blog sticker applied (if not, it can be done from the site's Blog RC).
* Insert a Donations block in a new post with the block editor.
* Verify you can toggle all the settings in the right sidebar and the UI updates accordingly.
* Click through the different tabs and makes sure that content displayed changes accordingly..
* Try editing the content of the inner blocks.
* Make sure your changes are preserved.
* Save the post and reload the editor.
* Confirm your changes are preserved.

#### To follow up

Items below are out of scope of this PR and will be implemented in follow-ups.

* Make currency editable.
* Make amounts editable (is this required for the MVP)?
* Save content into the post.
* Make text/background colors customizable (I guess this is not required for the MVP and can be implemented afterwards).

Part of #42856
